install.sh

```diff
@@ -1,116 +1,105 @@
 #!/usr/bin/env bash
 
 fancy_echo() {
-  local fmt="$1"; shift
-  printf "\n$fmt\n" "$@"
+  printf "\n%s\n" "$1"
 }
 
 fancy_echo "This script will setup your laptop"
 fancy_echo "If you want to reuse your old SSH key, copy your SSH config over before running this script"
-fancy_echo "During installation, it will ask for your sudo password a few times"
+fancy_echo "During installation, it may ask for your sudo password."
 
 # Get name and email
 fancy_echo "Before we start we need some basic details of you"
-read -p "What is your full name? (e.g. Johnny Appleseed): " full_name
-read -p "What is your email address? (e.g. johnny.appleseed@dockyard.com): " email_address
+read -r -p "What is your full name? (e.g. Johnny Appleseed): " full_name
+read -r -p "What is your email address? (e.g. johnny.appleseed@dockyard.com): " email_address
 
 fancy_echo "Hello $full_name <$email_address>"
 
 fancy_echo "Press enter to start the installation process, press <CTRL + C> to cancel"
-read
+read -r
 
-if [ -f ~/.ssh/id_rsa ]
+if [ -f ~/.ssh/id_ed25519 ]
 then
   fancy_echo "Skipping SSH key generation, you already have one"
 else
   fancy_echo "Generating SSH key"
-  ssh-keygen -q -t rsa -b 4096 -C "$email_address" -N "" -f ~/.ssh/id_rsa
+  ssh-keygen -q -t ed25519 -C "$email_address" -N "" -f ~/.ssh/id_ed25519
```

Comment:
Should we also skip generation if an `id_rsa` (of  >2048 bits) already exists? The original generates an RSA of sufficient strength to not need to be replaced with the newfangled ED25519.

you can check RSA key strength with:
```sh
ssh-keygen -lf ~/.ssh/id_rsa.pub | awk '{print $1;}'
```

Reply:
yes I think we should skip if id_rsa **or** ed25519 exists

Reply:
Now it will use id_rsa if it exists (and if id_ed25519 does not exist)
otherwise it will default to creating & using id_ed25519